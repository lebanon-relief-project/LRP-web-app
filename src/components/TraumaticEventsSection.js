import React from "react";

import styled from "styled-components";
import colours from "../styles/Colours";

const WhatHappened = () => {
    return (
        <StyledSection>
            <Wrapper>
                <StyledHeader>Traumatic Events</StyledHeader>
                <StyledParagraph>
                    The explosion is a traumatic event. It affected those that were in Beirut at the time, people all over
                    Lebanon, the Lebanese abroad, and those who know someone that was affected.

                    All these individuals, regardless of whether they were in Beirut at the time of the explosion or whether
                    they were directly impacted by the explosion can experience traumatic stress as a result of it.

                    Traumatic stress is changes in how one feels, thinks, or behaves as a result of trauma.
                    It is a normal reaction to an abnormal event.
                </StyledParagraph>
            </Wrapper>
        </StyledSection>
    );
};

const Wrapper = styled.div`
  width: 50%;
  padding: 5rem 1rem;
  margin: 0 12rem;
  background: inherit;
`;

const StyledSection = styled.section`
  background: ${colours.lightGrey};
  margin: 0;
  padding: 0;
`;

const StyledHeader = styled.h2`
  font-weight: bold;
`;

const StyledParagraph = styled.p`
  margin: 0;
`;

export default WhatHappened;