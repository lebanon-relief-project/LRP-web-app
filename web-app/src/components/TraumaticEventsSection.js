import React from "react";
import styled from "styled-components";
import colours from "../styles/Colours";
import devices from "../styles/Devices";
import HelpPageLeftBackground from "../assets/images/background/HelpPageLeftBackground.svg";
import HelpPageRightBackground from "../assets/images/background/HelpPageRightBackground.svg";

const TraumaticEvents = () => {
  return (
    <StyledSection>
      <HeaderWrapper>
        <HeaderLeftImageWrapper>
          <img src={HelpPageLeftBackground} alt={"background"} />
        </HeaderLeftImageWrapper>

        <HeaderRightImageWrapper>
          <img src={HelpPageRightBackground} alt={"background"} />
        </HeaderRightImageWrapper>
      </HeaderWrapper>

      <CentralWrapper>
        <WidthConstraint>
          <CentralContentWrapper>
            <StyledHeader>Traumatic Events</StyledHeader>
            <TextWrapper>
              <StyledParagraph>
                Trauma usually occurs after a deeply distressing or disturbing
                event, which may manifest as changes in feelings, thoughts, or
                behaviors. Trauma may also be less obvious, manifesting as a
                normal reaction to an abnormal event.
              </StyledParagraph>
              <StyledParagraph>
                It is important to keep in mind that trauma is unique to each
                individual. Any small or big distressing event can trigger it,
                and everyone experiences trauma differently. It can be
                collective and widespread, such as pandemics or economic crises,
                but it also can be individual and specific, such as abuse or the
                death of a loved one.
              </StyledParagraph>
              <StyledParagraph>
                One may also suffer from trauma without having experienced an
                event first hand, but rather by caring for someone who has or by
                simply watching the news.
              </StyledParagraph>
            </TextWrapper>
          </CentralContentWrapper>
        </WidthConstraint>
      </CentralWrapper>
    </StyledSection>
  );
};

const TextWrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;

  @media (max-width: ${devices.mobile}) {
    margin-bottom: 25px;
    margin-top: 15px;
  }
`;

const WidthConstraint = styled.div`
  width: inherit;
  max-width: 960px;
`;

const CentralWrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  z-index: 2;
  background-color: transparent;
  justify-content: center;
  position: absolute;
  align-items: center;
  height: 363px;
  height: 410px;
  width: 100%;

  @media (max-width: ${devices.mobile}) {
    height: 560px;
  }
`;

const CentralContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background-color: transparent;

  max-width: 632px;

  @media (max-width: ${devices.ipadpro}) {
    margin: 0 12.5%;
  }

  @media (max-width: ${devices.ipad}) {
    margin-left: 30px;
    margin-right: 30px;
  }

  @media (max-width: ${devices.mobile}) {
    margin-left: 18px;
    margin-right: 18px;
    flex: 1;
    margin-top: 20px;
  }
`;

const HeaderWrapper = styled.div`
  display: flex;
  flex: 1;
  z-index: 1;
  justify-content: space-between;
`;

const HeaderLeftImageWrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-start;
  align-items: center;

  @media (max-width: ${devices.ipadpro}) {
    display: none;
  }

  @media (max-width: ${devices.ipadprolandscape}) {
    opacity: 0.3;
  }
`;

const HeaderRightImageWrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-items: center;

  @media (max-width: ${devices.ipadpro}) {
    display: none;
  }

  @media (max-width: ${devices.ipadprolandscape}) {
    opacity: 0.3;
  }
`;

const StyledSection = styled.div`
  display: flex;
  background: ${colours.darkBlue};
  margin: 0;
  margin-top: 80px;
  padding: 0;
  width: 100vw;
  height: 410px;
  z-index: 0;

  @media (max-width: ${devices.ipad}) {
    margin-top: 80px;
    ${"" /* height: 560px */}
  }

  @media (max-width: ${devices.mobile}) {
    margin-top: 80px;
    height: 560px;
  }
`;

const StyledHeader = styled.h2`
  font-weight: bold;
  font-size: 38px;
  margin: 0;
  color: ${colours.white};
  @media (max-width: ${devices.mobile}) {
    font-size: 30px;
    line-height: 38px;
  }
`;

const StyledParagraph = styled.p`
  margin: 0;
  margin-bottom: 1rem;
  font-size: 16px;
  color: ${colours.white};
`;

export default TraumaticEvents;
