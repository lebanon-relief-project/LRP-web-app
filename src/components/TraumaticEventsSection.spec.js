import React from "react";
import { render } from "@testing-library/react";
import TraumaticEventsSection from "./TraumaticEventsSection";

describe("the TraumaticEvents component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderTraumaticEventsSection());

    expect(container).toMatchSnapshot();
  });
});

const renderTraumaticEventsSection = () => {
  return render(<TraumaticEventsSection />);
};
