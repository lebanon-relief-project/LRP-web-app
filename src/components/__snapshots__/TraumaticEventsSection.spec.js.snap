// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TraumaticEvents component should match the snapshot 1`] = `
<div>
  <section
    class="sc-gtssRu bgKeKC"
  >
    <div
      class="sc-bdnylx icjZAh"
    >
      <h2
        class="sc-dlnjPT iGXfwf"
      >
        Traumatic Events
      </h2>
      <p
        class="sc-hKFyIo hqFVHM"
      >
        The explosion is a traumatic event. It affected those that were in Beirut at the time, people all over Lebanon, the Lebanese abroad, and those who know someone that was affected. All these individuals, regardless of whether they were in Beirut at the time of the explosion or whether they were directly impacted by the explosion can experience traumatic stress as a result of it. Traumatic stress is changes in how one feels, thinks, or behaves as a result of trauma. It is a normal reaction to an abnormal event.
      </p>
    </div>
  </section>
</div>
`;
